ros2cli/ros2cli/node/strategy.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, args):
                 self._direct_node = DirectNode(args)
                 self._daemon_node = None
         else:
-            if use_daemon:
+            if use_daemon and getattr(args, 'spawn_daemon', True):
```

Comment:
This attribute seems a bit redundant; we already have the `use_daemon` attribute.  Can we just refactor this logic a bit instead?  Something like:

```
use_daemon = not getattr(args, 'no_daemon', False)
if use_daemon:
  if not is_daemon_running(args):
    spawn_daemon()
  # Do DaemonNode stuff
else:
  # Do DirectNode stuff
```

Reply:
It's good to see that we can get the consensus to skip spawning the daemon for the completer (As there is still a bug in it even in the foxy: After running the `ros2 topic echo /<tab>`, there is a _ros2_daemon respawned, but if `ctrl+c` in the current shell any time, the _ros2_daemon exit. This is not the daemon feature I expected).

> Something like:

I am confused with this refactor. There is a case:
If `ros2 daemon start` was run, do we expect the `ros2 topic echo /` can use `DaemonNode` to get data from `daemon`?

That's the reason why I added an extra `spawn_daemon` attribute.

Reply:
> After running the ros2 topic echo /<tab>, there is a _ros2_daemon respawned, but if ctrl+c in the current shell any time, the _ros2_daemon exit. This is not the daemon feature I expected

@iuhilnehc-ynos if i understand correctly, you are saying this is the behavior after your patch is applied? and i did not expect that behavior either.

Reply:
Yes. Even if the patch https://github.com/ros2/ros2cli/pull/852#issuecomment-1716847919 is applied.

Reply:
@clalancette Updated in https://github.com/ros2/ros2cli/pull/852/commits/aba8fe5abcb8eac7a2f7fee35e1324e04f0f622b